packages/aws-cdk/lib/api/cxapp/exec.ts

```diff
@@ -75,14 +77,40 @@ export async function execProgram(aws: SdkProvider, config: Configuration): Prom
   env[cxapi.OUTDIR_ENV] = outdir;
 
   // Send version information
-  env[cxapi.CLI_ASM_VERSION_ENV] = cxapi.CLOUD_ASSEMBLY_VERSION;
   env[cxapi.CLI_VERSION_ENV] = versionNumber();
 
   debug('env:', env);
 
   await exec();
 
-  return new cxapi.CloudAssembly(outdir);
+  return createAssembly(outdir);
+
+  function createAssembly(directory: string) {
+
+    const assembly = new cxapi.CloudAssembly(directory);
+    assertAssemblyVersion(assembly);
+    return assembly;
+
+  }
+
+  function assertAssemblyVersion(cloudAssembly: cxapi.CloudAssembly) {
+
+    function _parseVersion(version: string) {
+      const ver = semver.coerce(version);
+      if (!ver) {
+        throw new Error(`Could not parse "${version}" as semver`);
+      }
+      return ver;
+    }
+
+    const maximumAcceptedManifestSem = _parseVersion(cxschema.Manifest.version());
+    const manifestSem = _parseVersion(cloudAssembly.manifest.version);
+
+    if (semver.gt(manifestSem, maximumAcceptedManifestSem)) {
+      throw new Error(`Cloud assembly schema version mismatch: actual('${manifestSem}') > expected('${maximumAcceptedManifestSem}'). Consider upgrading your CLI version.`);
```

Comment:
```suggestion
      throw new Error(`error: cloud assembly schema version must be <= ${cxschema.Manifest.version()} but found ${cloudAssembly.manifest.version}. Please upgrade the CDK CLI`);
```

Reply:
The "error:" prefix is redundant. Exceptions are printed with "Error: <error-message>".

Reply:
I also think the "directionality" of the message should be ">=" and not "<=" (i.e. "the CLI version must be >= framework version")

Reply:
I'm ok with leaving the version mismatch msg as is.
But I would rephrase the 'upgrade' part of the message less ambiguously / more strongly.

Reply:
👍

Reply:
I've rephrased this. Take another look.